working-groups/meta/rfc-drafts/compiler-team-contributors.md

```diff
@@ -0,0 +1,244 @@
+Most notably:
+
+- you cannot be assigned to issues unless you have at least *read* access to a repository;
+- you cannot modify labels without *write* access (I think);
+- you cannot be a member of a *team*, which means you cannot be addressed via some
+  alias like `@rust-lang/compiler-team`;
+- you do not get the little "member" badge appearing next to your name when you comment.
+
+The last point is potentially important: by being made a member of the
+org, you are to some extent representing that org, as you are
+visibility identified as a member.  These can be important in terms of
+the code of conduct, as we wish for representatives of rust-lang to
+take extra care in their public interaction. In particular, this
+implies we might not want to allow **anyone** to make themselves a
+member of the org.
+
+### triagebot
+
+The triagebot is an "upcoming" piece of infrastructure that should allow any GitHub user
+to make some changes to issues on rust-lang repositories. In particular, one would be
+able to instruct the triagebot to do the following:
+
+- adjust labels on issues
+- assign oneself to the issue
+
+Because the triagebot can be used by anyone, and not just org members, assigning
+works as follows:
+
+- the issue is *officially* assigned to the triagebot (as far as
+  Github is concerned, that is)
+- the issue header is edited to indicate that it is assigned to the
+  user in question
+
+This is a bit less good than being assigned to the issue as an org
+member, since it means that your username and picture do not appear
+next to the issue, but it's still pretty decent and should suffice for
+most purposes.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## The path to membership
+
+People will typically start as a **working group participant,** which
+is basically somebody who has come to work on something for the first
+time. They don't know much about the compiler yet and have no
+particular privileges. They are assigned to issues using the triagebot
+and (typically) work with a mentor or mentoring instructions.
+  
+Once a working group participant has been contributing regularly for
+some time, they can be promoted to the level of a **compiler team
+contributor**. This indicates that they are someone who contributes
+regularly.
+
+- Contributors have r+ privileges and can do reviews (they are
+  expected to use those powers appropriately, as discussed
+  previously).
+- Contributors are members of the organization so they can modify
+  labels and be assigned to issues.
+- Contributors will be asked if they wish to be added to highfive rotation.
+- Contributors will be listed on the [compiler expert map](https://github.com/rust-lang/compiler-team/blob/9d8c387ddbd01ced14eaab480cddb00c2d723f36/experts/MAP.md),
+  which lists folks who are familiar with each area of the compiler.
+- Contributors are listed on the rust-lang.org web page and invited to
+  the Rust All Hands.
```

Comment:
Perhaps being able to lead/co-lead a working group could be another thing to list here. I don't know that we have any hard requirements on who can do this right now but I think it would make sense that we'd want working group leads to be people that the team can rely on and trust.

Reply:
Yeah. This reminds me that I actually think it would make sense to try and write an RFC **for working groups themselves**. e.g., one question might be "who can lead a WG"? My impression is that WGs should have a member as a lead, but could (and should!) have co-leads who are not members.

Reply:
Does "not members" in this context mean "compiler team contributors" (from this PR) or anyone who isn't a compiler team member?

Reply:
I meant contributors.

Reply:
I was mainly thinking of compiler team members as the ones that lead a WG more than compiler team contributors. I guess it depends on the WG, but if the WG is about a huge compiler area and someone is leading that is because the person has deep knowledge and hence it's a compiler team member, or am I wrong?.

Reply:
> I guess it depends on the WG, but if the WG is about a huge compiler area and someone is leading that is because the person has deep knowledge and hence it's a compiler team member, or am I wrong?.

This is my understanding also, but I think having a compiler team contributor co-lead could be a good opportunity for them to learn and take on new responsibilities (perhaps as part of the active mentoring, mentioned toward the end of the document).

Reply:
To be honest, I think the term lead is ungreat -- or maybe I should say, the *role* of lead is ungreat. I think there might be value in trying to divvy up the responsibilities in a more fine-grained fashion. For example:

- Running the meeting
- Taking and posting minutes from the meeting
- Technical design and leadership
- Writing mentoring instructions
- Publicity and recruiting -- blog posts, tweets, calling attention to what's going on
- Synchronizing with the team at large to give status updates etc
- Being a "point of contact" for questions

I think that all of these are things that the leads have traditionally done, but I think they could easily be separated out. It makes sense to me that the compiler team members would be needed for technical design and leadership -- but I don't think it's exclusive to compiler team members. Think of the work @matthewjasper did on altering the match desugaring and handling 2PB in the NLL group. 

From another perspective, speaking as someone who often winds up playing the lead, I can say that it is an enormous relief when (for example) somebody else takes the reins of "running meetings" and making sure they happen. It's  a small thing, but it makes a difference. 

Anyway, I'm not sure where these thoughts belong -- in this RFC? Somewhere else? In a blogpost?

Reply:
@nikomatsakis 100% agreed. Maybe it worth mentioning this in this RFC but also explaining this in meta WG in the documents of forming a new WG and also maybe as a general thing for the compiler team in the compiler repo because this applies not only to working groups but also to leading specific parts of the compiler.

Reply:
Added a throw-away sentence, we'll come back to this